skglm/penalties/non_separable.py

```diff
@@ -40,14 +40,16 @@ def value(self, w):
         return np.sum(np.sort(np.abs(w)) * self.alphas[::-1])
 
     def prox_vec(self, x, stepsize):
-        def _prox(_x, _alphas):
-            sign_x = np.sign(_x)
-            _x = np.abs(_x)
-            sorted_indices = np.argsort(_x)[::-1]
-            prox = prox_SLOPE(_x[sorted_indices], _alphas)
-            prox[sorted_indices] = prox
-            return prox * sign_x
-        return _prox(x, self.alphas * stepsize)
+        _alphas = self.alphas * stepsize
+        _x = x
+        prox = np.zeros_like(x)
+
+        sign_x = np.sign(_x)
+        _x = np.abs(_x)
+        sorted_indices = np.argsort(_x)[::-1]
+        prox[sorted_indices] = prox_SLOPE(_x[sorted_indices], _alphas)
```

Comment:
```suggestion
        alphas = self.alphas * stepsize
        prox = np.zeros_like(x)

        sign_x = np.sign(x)
        x = np.abs(x)
        sorted_indices = np.argsort(x)[::-1]
        prox[sorted_indices] = prox_SLOPE(x[sorted_indices], alphas)
```

Reply:
@Badr-MOUFAD LGTY?

Reply:
LGTM!
one small remark, I changed 
```python
_alphas = self.alphas * stepsize
```
to avoid confusion

Reply:
better indeed, thanks!